pkg/commands/execution_context.go

```diff
@@ -1487,37 +1487,49 @@ func (ctx *executionContext) findShellTest(exec *config.Execution) []*model.Test
 
 func (ctx *executionContext) findGoTest(executionConfig *config.Execution) ([]*model.TestEntry, error) {
 	st := time.Now()
+
 	logrus.Infof("Starting finding tests by source %v", executionConfig.Source)
+
 	execTests, err := model.GetTestConfiguration(ctx.manager, executionConfig.PackageRoot, executionConfig.Source)
 	if err != nil {
 		logrus.Errorf("Failed during test lookup %v", err)
 		return nil, err
 	}
+
 	result, err := ctx.findGoSuites(executionConfig, execTests)
 	if err != nil {
 		return nil, errors.Wrapf(err, "an error during searching go suites")
 	}
-	logrus.Infof("Tests found: %v Elapsed: %v", len(execTests), time.Since(st))
+
+	testCount := len(execTests)
+	for _, testEntry := range result {
+		testCount += len(testEntry.Suite.Tests)
+	}
+
+	logrus.Infof("Tests found: %v Elapsed: %v", testCount, time.Since(st))
+
 	for _, t := range execTests {
 		t.Kind = model.GoTestKind
 		t.ExecutionConfig = executionConfig
 		if len(executionConfig.OnlyRun) == 0 || utils.Contains(executionConfig.OnlyRun, t.Name) {
 			result = append(result, t)
+		} else {
+			testCount--
 		}
 	}
-	if len(result) != len(execTests) {
-		logrus.Infof("Tests after filtering: %v", len(result))
-	}
+
+	logrus.Infof("Tests after filtering: %v", testCount)
```

Comment:
Why do we need a write about filter even if the test count is not changed?

Reply:
Fixed.